config/core/resources/channel.yaml

```diff
@@ -74,7 +74,8 @@ spec:
                     description: Spec defines the Spec to use for each channel
                         created. Passed in verbatim to the Channel CRD as Spec
                         section.
-                    type: string
+                    type: object
+                    x-kubernetes-preserve-unknown-fields: true
```

Comment:
look, that was also removed, in between https://github.com/knative/eventing/commit/31ef0946172e9bafef7235eb7f3b51aa5b0ece69#diff-0231d3c043c857ee3a2e2f93630b183580a12e09d21affb1b1970f483e5c0bd8L55

Reply:
Now we have a test for it, and shouldn't happen again. :)
I see a commit below that references this PR, are you backporting it? Otherwise, I can do it.